miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt

```diff
@@ -2,29 +2,62 @@ package com.rakuten.tech.mobile.miniapp.js
 
 import android.webkit.JavascriptInterface
 import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
 import com.rakuten.tech.mobile.miniapp.display.WebViewListener
 
+@Suppress("TooGenericExceptionCaught", "SwallowedException")
 /** Bridge interface for communicating with mini app. **/
 abstract class MiniAppMessageBridge {
     private lateinit var webViewListener: WebViewListener
 
     /** Get provided id of mini app for any purpose. **/
     abstract fun getUniqueId(): String
 
+    /** Post permission request from external. **/
+    abstract fun requestPermission(
+        callbackId: String,
+        miniAppPermissionType: String,
+        permissions: Array<String>
```

Comment:
I don't think we need to pass an Array of permissions. It should be enough to pass a custom type such as `MiniAppPermissionType.GeoLocation` and then the Host App is responsible for interpreting which device permissions they need to request. That will let the Host App have more flexibility over the permissions and simplify things in the SDK.